### DISCUSSION TITLE
Table inside div is not compiling correctly due to indent

### DISCUSSION THREAD

**Discussion by hahnbeelee** [mdx-js/mdx#2245] [Q&A] [answered]
I have a remarkGfm-formatted table embedded inside a div, but unfortunately it doesn't compile properly. I run my mdx through a pre-parsing step that I have replicated [in this sandbox](https://codesandbox.io/s/table-in-div-lq11of?file=/src/index.js). If you see the before and after, you can see that the code inside the div got indented after the parsing step. Is there any way to prevent this? I discovered that removing the indent displays the table properly.

With indent:
![image](https://user-images.githubusercontent.com/55263191/217114068-972815e1-12ac-44ab-9983-3c7bbc08e9eb.png)
![image](https://user-images.githubusercontent.com/55263191/217114266-6bae77bd-0e2a-4590-93d0-b34dbd8fad83.png)

Without indent:
![image](https://user-images.githubusercontent.com/55263191/217114403-4022ad16-7c0c-458a-8549-4761d52fbaa0.png)
![image](https://user-images.githubusercontent.com/55263191/217114364-532af917-0db1-4657-8615-0f5679308246.png)

**Comment by wooorm** [comment]
What do you mean it doesn’t compile correctly?
The indent is intended: in MDX you can indent things.
Perhaps the place you screenshot from doesn’t support MDX?

**Reply by hahnbeelee** [reply]
I dont want the indent, if I remove the indent it compiles the table.

**Reply by wooorm** [reply]
What is “it”? Where is it?

**Reply by hahnbeelee** [reply]
But when I run it through remark it automatically indents it.

**Reply by hahnbeelee** [reply]
I'm using hashicorp's next-mdx-remote inside my next.js app.

**Reply by wooorm** [reply]
Yes, remark indents it.
`next-mdx-remote` supports MDX 2 as far as I understand, so it should support this too?

**Reply by hahnbeelee** [reply]
I see. To me it feels like an MDX parsing error where the GFM table is not being properly detected as a table due to the indent.

**Reply by hahnbeelee** [reply]
Re-reading back if you also agree with me that this is the case I will close this discussion and create an issue on the `remark-gfm` package repo.

**Reply by hahnbeelee** [reply]
Sigh just kidding. Thank you for rubber ducky-ing https://codesandbox.io/s/gfm-indent-test-qc5ejy?file=/src/index.js
I tried this and it worked, so I'm not sure what the issue is with it not rendering. I will have to debug the intricacies of my set up...

**Reply by wooorm** [reply]
Haha!
Yeah, we support this fine. But it’s something with what you use to render to HTML to produce that screenshot.
I think that *that* only supports markdown, not MDX? Or it supports MDX 1?

**Reply by hahnbeelee** [reply]
I think we're on MDX 2. Maybe I'll try bumping packages just in case.

**Reply by ghoshnirmalya** [reply]
The issue happens when there is more than a one-level indentation for tables.

In [this CodeSandbox](https://codesandbox.io/s/gfm-indent-test-forked-3p6x58?file=/src/index.js:802-1006), you can notice that the first table renders fine with the expected HTML. However, the second table contains a two-level indentation and it renders as a string.

The markdown code for the above is:

```mdx
<div>

  Required parameters for integration:

  | Parameter | Description                    |
  | --------- | ------------------------------ |
  | Domain    | domain description             |
  | Api Key   | Secret API Access Key.         |

  <div>

    | Parameter | Description                    |
    | --------- | ------------------------------ |
    | Domain    | domain description             |
    | Api Key   | Secret API Access Key.         |

  </div>
</div>
```

The HTML that the above markdown generates is as follows:
```html
<div><p>Required parameters for integration:</p><table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>Domain</td>
<td>domain description</td>
</tr>
<tr>
<td>Api Key</td>
<td>Secret API Access Key.</td>
</tr>
</tbody>
</table><div><p>| Parameter | Description                    |
| --------- | ------------------------------ |
| Domain    | domain description             |
| Api Key   | Secret API Access Key.         |</p></div></div>
```

**Reply by hahnbeelee** [reply]
hey @wooorm I think this is a bug in the parser, but would love your insight.

**Reply by wooorm** [reply, marked as answer]
definitely possible! might also already be fixed in https://github.com/micromark/micromark-extension-gfm-table/releases/tag/1.0.6, have you deleted `node_modules` and tried reinstalling? Otherwise, that’s the project where the bug goes.
There’s a related test here: https://github.com/micromark/micromark-extension-gfm-table/blob/f5e8ac19d01ee5c6ad1113c3fec1f44e205450bf/test/index.js#L367-L378

**Reply by ghoshnirmalya** [reply]
@wooorm The above solution worked for us. Thanks.

Note that the dependencies are updated in the [remark-gfm](https://github.com/remarkjs/remark-gfm) repository. I'm happy to do that.

**Reply by wooorm** [reply]
Which solution? Removing node_modules?
Nothing needs to update in `remark-gfm` AFAIK, it uses semver to pull in bug fixes: if the bug fixes were not pulled in for you, that’s because you explicitly choose not to get bug fixes.

**Reply by ghoshnirmalya** [reply]
What I meant is that in `remark-gfm`, the version of [`micromark-extension-gfm` is ^2.0.0](https://github.com/remarkjs/remark-gfm/blob/main/package.json#L42) whereas [the latest version of `micromark-extension-gfm` is 3.0.0](https://github.com/micromark/micromark-extension-gfm/releases).

**Reply by wooorm** [reply]
That version does not include fixes/changes to the algorithm, just more generic updates on how it can be used (https://github.com/micromark/micromark-extension-gfm/releases/tag/3.0.0). It takes a while to update all packages in the unified ecosystem. Will get to it soon.
But it won’t affect this issue!
